src/StringTools/InternableString.cs

```diff
@@ -265,6 +265,15 @@ public unsafe string ExpensiveConvertToString()
                         }
                     }
                 }
+
+                // The invariant that Length is the sum of span lengths is critical in this unsafe method.
+                // Violating it may lead to memory corruption and, since this code tends to run under a lock,
+                // to hangs caused by the lock getting orphaned. Attempt to detect that and throw now, 
+                // before the corruption causes further problems.
+                if (destPtr != resultPtr + Length)
+                {
+                    throw new InvalidOperationException("Length property does not match sum of span lengths");
```

Comment:
maybe also print the values of all the three numbers if it will make debugging easier?

Reply:
Yes, I have added two numbers to the exception message. The pointer value is not of much use without a dump and can be easily extracted from the dump if it's available. Thank you!